src/relay/pass/type_solver.cc

```diff
@@ -38,6 +39,194 @@ class TypeSolver::Reporter : public TypeReporterNode {
   TypeSolver* solver_;
 };
 
+class TypeSolver::RecurrenceChecker : public TypeVisitor {
```

Comment:
Can we name this occurs check as the literature does?

Reply:
Will do, thanks